### PR TITLE
Separate fetching a single identity and fetching many in DataSource

### PR DESCRIPTION
**Pull request by dialelo**
@raulraja please take a look. @MaureenElsberry if you could do a language pass to the edited docs would be great.

### CODE REVIEW

File: docs/src/tut/docs.md [hunk 1]
```diff
@@ -123,6 +134,23 @@ given an id, we just have to pass a `UserId` as an argument to `Fetch`.
 def getUser(id: UserId): Fetch[User] = Fetch(id) // or, more explicitly: Fetch(id)(UserSource)
 ```
 
+
+### Data sources that don't support batcihng
```
Comment (raulraja) [inline docs/src/tut/docs.md]: Batching

File: shared/src/main/scala/datasource.scala [hunk 2]
```diff
@@ -19,12 +19,48 @@ package fetch
...
+    */
+  def fetchMany(ids: NonEmptyList[I]): Eval[Map[I, A]]
+
+  /** Use `fetchOne` for implementing of `fetchMany`. Use only when the data
+    * source doesn't support batching.
+    */
+  def batchingNotSupported(ids: NonEmptyList[I]): Eval[Map[I, A]] = {
+    val idsList = ids.unwrap
+    idsList
+      .map(fetchOne)
```
Comment (raulraja) [inline shared/src/main/scala/datasource.scala]: When we use Task here we can paralelize this with a strategy argument

File: shared/src/test/scala/FetchTests.scala [hunk 3]
```diff
@@ -34,31 +34,39 @@ object TestHelper {
...
   implicit object ManySource extends DataSource[Many, List[Int]] {
     override def name = "ManySource"
-    override def fetch(ids: NonEmptyList[Many]): Eval[Map[Many, List[Int]]] =
+    override def fetchOne(id: Many): Eval[Option[List[Int]]] =
+      M.pure(Option(0 until id.n toList))
+    override def fetchMany(ids: NonEmptyList[Many]): Eval[Map[Many, List[Int]]] =
       M.pure(ids.unwrap.map(m => (m, 0 until m.n toList)).toMap)
   }
 
   case class Never()
```
Comment (raulraja) [inline shared/src/test/scala/FetchTests.scala]: We should make all case classes final

**Comment by raulraja** [comment]
LGTM!

File: shared/src/main/scala/datasource.scala (hunk 2 above)
Comment (dialelo) [inline shared/src/main/scala/datasource.scala]: Working on that on a separate branch, will take this into account!

**Pull request merged**